# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLChild.cpp
# Commit: ccfa767dba64
# Full Hash: ccfa767dba644dc193e0246eb8e8ff3377e8b8a5
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Client-side bindings mirror for precise CC, and merge similar codepaths. r=handyman
#   
#   * Context loss using RAII
#   * Move Program reflection Client-side
#   
# ==============================================================================

diff -r 71c122ac0ca7 -r ccfa767dba64 dom/canvas/WebGLChild.cpp
--- a/dom/canvas/WebGLChild.cpp	Wed Jan 08 22:19:14 2020 +0000
+++ b/dom/canvas/WebGLChild.cpp	Wed Jan 08 22:19:16 2020 +0000
@@ -6,31 +6,24 @@
 #include "WebGLChild.h"
 
 #include "ClientWebGLContext.h"
-#include "nsICanvasRenderingContextInternal.h"
-#include "nsQueryObject.h"
 
 namespace mozilla {
-
 namespace dom {
 
-mozilla::ipc::IPCResult WebGLChild::RecvQueueFailed() {
-  mozilla::ClientWebGLContext* context = GetContext();
-  if (context) {
-    context->OnQueueFailed();
-  }
-  return Send__delete__(this) ? IPC_OK() : IPC_FAIL_NO_REASON(this);
+WebGLChild::WebGLChild(ClientWebGLContext& context) : mContext(context) {}
+
+WebGLChild::~WebGLChild() { (void)Send__delete__(this); }
+
+mozilla::ipc::IPCResult WebGLChild::RecvJsWarning(
+    const std::string& text) const {
+  mContext.JsWarning(text);
+  return IPC_OK();
 }
 
-mozilla::ClientWebGLContext* WebGLChild::GetContext() {
-  nsCOMPtr<nsICanvasRenderingContextInternal> ret = do_QueryReferent(mContext);
-  if (!ret) {
-    return nullptr;
-  }
-  return static_cast<mozilla::ClientWebGLContext*>(ret.get());
-}
-
-void WebGLChild::SetContext(mozilla::ClientWebGLContext* aContext) {
-  mContext = do_GetWeakReference(aContext);
+mozilla::ipc::IPCResult WebGLChild::RecvOnContextLoss(
+    const webgl::ContextLossReason reason) const {
+  mContext.OnContextLoss(reason);
+  return IPC_OK();
 }
 
 }  // namespace dom